### PR TITLE
update(logging): Strip ansi from logs and general logging improvements

### PR DESCRIPTION
**Pull request by ShanaryS**
Logging improvements that were made across branches. Also added a new dependency strip-ansi from the same maintainers as chalk. This removes the ansi codes that clutters up the logs but keeps them in console. Also redacting more sensitive information.

### CODE REVIEW

File: src/pipeline.ts [hunk 1]
```diff
@@ -161,16 +164,21 @@ async function findMatchesBatch(
 	const { delay } = getRuntimeConfig();
 
 	let totalFound = 0;
-	for (const [i, sample] of samples.entries()) {
+	for (const [i, searchee] of samples.entries()) {
+		const searcheeLog = searchee.infoHash
+			? `${chalk.bold.white(searchee.name)} ${chalk.dim(`[${searchee.infoHash.slice(0, 8)}...]`)}`
+			: searchee.path
+				? chalk.bold.white(searchee.path)
+				: chalk.bold.white(searchee.name);
```
Comment (mmgoodnow) [inline src/pipeline.ts:172]: can this be extracted into a helper?

File: src/utils.ts [hunk 2]
```diff
@@ -129,6 +129,8 @@ export function getAnimeQueries(name: string): string[] {
 		animeQueries.push(cleanseSeparators(`${title} ${release}`));
 	}
 	if (altTitle) {
+		if (altTitle.toLowerCase() === "season") return animeQueries;
+		if (altTitle.toLowerCase() === "ep") return animeQueries;
```
Comment (mmgoodnow) [inline src/utils.ts:133]: why this change?

File: src/torznab.ts [hunk 3]
```diff
@@ -681,3 +662,76 @@ async function makeRequests(
...
+		skippingIndexersCallout +=
+			indexersToUseByTimestamp.length === indexersToUse.length
+				? "timestamps)"
+				: "timestamps ";
+	}
+	if (indexersToUseByTimestamp.length > indexersToUse.length) {
+		skippingIndexersCallout +=
+			enabledIndexers.length === indexersToUseByTimestamp.length
+				? "supported category)"
+				: "and supported category)";
```
Comment (mmgoodnow) [inline src/torznab.ts:710]: there's a util for this: https://github.com/cross-seed/cross-seed/blob/9558b31f908d731d184c2d2b17b983175aab1d69/src/utils.ts#L163

File: src/torznab.ts [hunk 4]
```diff
@@ -681,3 +662,76 @@ async function makeRequests(
...
+	}
+	if (indexersToUse.length === 0) {
+		logger.info({
+			label: Label.TORZNAB,
+			message: `${skippingSearching(skippingIndexersCallout)}`,
+		});
+		throw new Error("SKIPPED");
```
Comment (mmgoodnow) [inline src/torznab.ts:717]: I don't really like the use of throw/catch for control flow like this. is there a way to achieve this behavior without throw/catch?

File: src/torznab.ts [hunk 5]
```diff
@@ -681,3 +662,76 @@ async function makeRequests(
...
+	const res: Result<ParsedMedia, boolean> =
+		indexersToUse.length > 0
+			? await scanAllArrsForMedia(searchee, mediaType)
+			: resultOfErr(false);
+	const parsedMedia = res.isOk() ? res.unwrap() : undefined;
+	const ids = parsedMedia?.movie ?? parsedMedia?.series;
+	const idsStr = ids ? `: ${formatFoundIds(ids)}` : "";
+
+	const skippingIndexers = `Searching ${indexersToUse.length} indexers for ${searchee.name}${skippingIndexersCallout} [${mediaTypeStr}${idsStr}]`;
+	logger.info({
```
Comment (mmgoodnow) [inline src/torznab.ts:730]: this log line contains a lot of information. can some of it move to a followup verbose log?

File: src/utils.ts (hunk 2 above)
Comment (ShanaryS) [reply src/utils.ts:133]: Some of my anime was parsing these as alternate titles due to weird naming. It would end up searching "season" or "ep". It's easier to reject these here instead of through the regex.

File: src/torznab.ts (hunk 5 above)
Comment (ShanaryS) [reply src/torznab.ts:730]: I like having these all in info so it's easier for users to catch something being wrong. I myself have caught bugs because something didn't parse as expected. If we are to split it up, I rather put `indexersToUse.length` and `skippingIndxersCallout` in verbose.

File: src/logger.ts [hunk 6]
```diff
@@ -53,8 +54,8 @@ function redactMessage(message: string | unknown, options?: RuntimeConfig) {
 	const runtimeConfig = options ?? getRuntimeConfig();
 	let ret = message;
 
-	// redact torznab api keys
-	ret = ret.replace(/apikey=[a-zA-Z0-9]+/g, `apikey=${redactionMsg}`);
+	ret = ret.replace(/key=[a-zA-Z0-9]+/g, `key=${redactionMsg}`);
+	ret = ret.replace(/pass=[a-zA-Z0-9]+/g, `pass=${redactionMsg}`);
```
Comment (zakkarry) [inline src/logger.ts:58]:
```suggestion
	ret = ret.replace(/pass=[a-zA-Z0-9]+/g, `pass=${redactionMsg}`);
	ret = ret.replace(/(?:auto\.\d+\.)([a-zA-Z0-9]+)/g, (match, key) =>
		match.replace(key, redactionMsg),
	);
```
needed for a specific tracker's custom formatting. tested already.

**Review by zakkarry — CHANGES_REQUESTED** [review]

File: src/logger.ts (hunk 6 above)
Comment (mmgoodnow) [reply src/logger.ts:58]: what format does it take?

File: src/torznab.ts (hunk 5 above)
Comment (mmgoodnow) [reply src/torznab.ts:726]: sg

File: src/utils.ts (hunk 2 above)
Comment (mmgoodnow) [reply src/utils.ts:135]: can you add a comment to that effect?

File: src/logger.ts (hunk 6 above)
Comment (zakkarry) [reply src/logger.ts:58]: it appends the rss key to the URL in the format

tracker.com/download/auto.torrentid#.rsskey

**Review by zakkarry — APPROVED** [review]

**Comment by ShanaryS** [comment]
Removed the logging for searching from pipeline.ts so only torznab.ts logs.

File: src/torznab.ts [hunk 7]
```diff
@@ -681,3 +642,98 @@ async function makeRequests(
...
+				: "and supported category)",
+		);
+	}
+	const reasonStr = new Intl.ListFormat("en", {
+		style: "narrow",
+		type: "unit",
+	}).format(skippingIndexersCallout);
+	if (indexersToUse.length === 0) {
+		logger.info(`${skippingSearching(reasonStr)}`);
+		return { indexersToUse };
```
Comment (mmgoodnow) [inline src/torznab.ts:721]: this is not really taking advantage of Intl.ListFormat, which if used idiomatically, will handle all of the things you're doing with ternaries.

Try this: 
```js
const filteringCauses = [
  areIndexersBeingFilteredByTimestamps,
  areIndexersBeingFilteredBySupportedCategory,
].filter(isTruthy);
const reasonStr = filteringCauses.length ? `(filtered by ${formatAsList(filteringCauses)})` ? '';
```

File: src/utils.ts [hunk 8]
```diff
@@ -160,6 +166,17 @@ export function humanReadableDate(timestamp: number): string {
 	return new Date(timestamp).toLocaleString("sv");
 }
 
+export function getLogString(data: Metafile | Searchee, color: ChalkInstance) {
+	if (data instanceof Metafile) {
+		return `${color(data.name)} ${chalk.dim(`[${data.infoHash.slice(0, 8)}...]`)}`;
+	}
```
Comment (mmgoodnow) [inline src/utils.ts:172]: is this instanceof check actually necessary? I would think duck typing should just work here

Comment (ShanaryS) [reply src/utils.ts:172]: It won't handle data.path without this check. It still thinks it could be Searchee after.

**Review by mmgoodnow — DISMISSED** [review]
